core/runtime/src/main/java/io/quarkus/runtime/util/ExceptionUtil.java

```diff
@@ -76,6 +76,16 @@ public static String rootCauseFirstStackTrace(final Throwable exception) {
         return generateStackTrace(modifiedRoot).replace("Caused by:", "Resulted in:");
     }
 
+    public static Throwable getRootCause(Throwable exception) {
+        final List<Throwable> exceptionChain = new ArrayList<>();
+        Throwable curr = exception;
```

Comment:
That looks like the list is not useful, and it feels weird that if the exception has no cause it's not considered the root cause, since `null` will be returned.

Reply:
It should only return `null` if the `exception` is `null`, otherwise the list if never empty... However, we should handle the recursive cause structures as well. I'll fix it and add a test.

Reply:
Done.